File: margyam-ai/src/common/collection_list_provider.py
class CollectionListProvider:
    @staticmethod
    def parse_query_params(params: dict) -> dict:
        page = int(params.get("page", 1)) if str(params.get("page", 1)).isdigit() else 1
        limit = int(params.get("limit", 20)) if str(params.get("limit", 20)).isdigit() else 20
        return {
            "page": max(1, page),
            "limit": min(100, max(1, limit)),
            "query": params.get("q", ""),
        }

File: margyam-gateway/src/common/CollectionListProvider.js
/**
 * CollectionListProvider
 * Generic pagination and query parameters parsing provider.
 */
export class CollectionListProvider {
  static parseQueryParams(query = {}) {
    const { page, limit, sortBy, sort, q, ...filters } = query;

    const parsedOptions = {
      page: parseInt(page, 10) > 0 ? parseInt(page, 10) : 1,
      limit: parseInt(limit, 10) > 0 ? parseInt(limit, 10) : 20,
      sort: {},
    };

    if (sortBy) {
      const sortOrder = String(sort).toLowerCase() === 'asc' ? 1 : -1;
      parsedOptions.sort[sortBy] = sortOrder;
    }

    return {
      options: parsedOptions,
      searchQuery: q || '',
      filters: this.sanitizeFilters(filters),
    };
  }

  static sanitizeFilters(filters) {
    const cleaned = {};
    for (const [key, value] of Object.entries(filters)) {
      if (value === undefined || value === null || value === '') continue;
      cleaned[key] = value;
    }
    return cleaned;
  }
}
